src/c1_foeppl_von_karman/foeppl_von_karman_curvable_bell_elements.h

```diff
@@ -0,0 +1,1300 @@
+                                                const unsigned& k_type) const
+    {
+      unsigned index = w_field_index();
+      return CurvableBellElement<NNODE_1D>::internal_value_for_field_of_type(
+        t_time, index, k_type);
+    }
+
+
+  protected:
+    /// In-plane basis functions and derivatives w.r.t. global
+    /// coords at local coordinate s; return det(Jacobian of mapping)
+    virtual double dbasis_u_eulerian_foeppl_von_karman(const Vector<double>& s,
+                                                       Shape& psi_n,
+                                                       DShape& dpsi_n_dx) const;
+
+    /// In-plane basis/test functions at and derivatives w.r.t
+    /// global coords at local coordinate s; return det(Jacobian of mapping)
+    virtual double dbasis_and_dtest_u_eulerian_foeppl_von_karman(
+      const Vector<double>& s,
+      Shape& psi_n,
+      DShape& dpsi_n_dx,
+      Shape& test_n,
+      DShape& dtest_n_dx) const;
+
+    /// Out-of-plane basis/test functions at local coordinate s
+    virtual void basis_and_test_w_foeppl_von_karman(const Vector<double>& s,
+                                                    Shape& psi_n,
+                                                    Shape& psi_i,
+                                                    Shape& test_n,
+                                                    Shape& test_i) const;
+
+    /// Out-of-plane basis/test functions and first derivs w.r.t.
+    /// to global coords at local coordinate s; return det(Jacobian of mapping)
+    virtual double dbasis_and_dtest_w_eulerian_foeppl_von_karman(
+      const Vector<double>& s,
+      Shape& psi_n,
+      Shape& psi_i,
+      DShape& dpsi_n_dx,
+      DShape& dpsi_i_dx,
+      Shape& test_n,
+      Shape& test_i,
+      DShape& dtest_n_dx,
+      DShape& dtest_i_dx) const;
+
+    /// Out-of-plane basis functions and derivs w.r.t. global
+    /// coords at local coordinate s; return det(Jacobian of mapping)
+    virtual double d2basis_w_eulerian_foeppl_von_karman(
+      const Vector<double>& s,
+      Shape& psi_n,
+      Shape& psi_i,
+      DShape& dpsi_n_dx,
+      DShape& dpsi_i_dx,
+      DShape& d2psi_n_dx2,
+      DShape& d2psi_i_dx2) const;
+
+    /// Out-of-plane basis/test functions and first/second derivs
+    /// w.r.t. to global coords at local coordinate s;
+    /// return det(Jacobian of mapping)
+    virtual double d2basis_and_d2test_w_eulerian_foeppl_von_karman(
+      const Vector<double>& s,
+      Shape& psi_n,
+      Shape& psi_i,
+      DShape& dpsi_n_dx,
+      DShape& dpsi_i_dx,
+      DShape& d2psi_n_dx2,
+      DShape& d2psi_i_dx2,
+      Shape& test_n,
+      Shape& test_i,
+      DShape& dtest_n_dx,
+      DShape& dtest_i_dx,
+      DShape& d2test_n_dx2,
+      DShape& d2test_i_dx2) const;
+
+    // End of FoepplVonKarmanEquations interface functions
+    //----------------------------------------------------------------------------
+
+  public:
```

Comment:
jumping between private, public and protected isn't nice. however, to facilitate tracking of changes probably best to leave these for now.